lib/iris/_lazy_data.py

```diff
@@ -102,6 +102,33 @@ def as_lazy_data(data, chunks=None, asarray=False):
     return data
 
 
+def _co_realise_lazy_arrays(arrays):
+    """
+    Compute multiple lazy arrays and return a list of real values.
+
+    All the arrays are computed together, so they can share results for common
+    graph elements.
+
+    Also converts any MaskedConstants returned into masked arrays, to ensure
+    that all return values are writeable NumPy array objects.
+
+    """
+    computed_arrays = da.compute(*arrays)
+    results = []
+    for lazy_in, real_out in zip(arrays, computed_arrays):
+        real_out = np.asanyarray(real_out)
+        if isinstance(real_out, ma.core.MaskedConstant):
+            # Convert any masked constants into NumPy masked arrays :  In some
+            # cases dask may return a scalar numpy.int/numpy.float object
+            # rather than a numpy.ndarray object.
+            # Recorded in https://github.com/dask/dask/issues/2111.
```

Comment:
The above comment isn't related to MaskedConstants specifically, but rather the (lack of) distinction in NumPy between scalar values (e.g. `np.float64(1.)`) and scalar arrays (e.g. `np.array(1., dtype=np.float64)`). It refers to the `asanyarray()` at line 119.

Reply:
Yes, I though so, but frankly I wasn't totally certain !
This comment is basically cut+paste from what was formerly in the as_concrete_data function.
I will amend ...